src/Forms/DefaultFormFactory.php

```diff
@@ -96,15 +97,11 @@ protected function getFormActions(RequestHandler $controller = null, $name, $con
      */
     protected function getFormValidator(RequestHandler $controller = null, $name, $context = [])
     {
-        $validator = null;
-        if ($context['Record']->hasMethod('getCMSValidator')) {
-            // @todo Deprecate or formalise support for getCMSValidator()
-            $validator = $context['Record']->getCMSValidator();
+        if (!$context['Record'] instanceof DataObject) {
+            return null;
         }
 
-        // Extend validator
-        $this->invokeWithExtensions('updateFormValidator', $validator, $controller, $name, $context);
```

Comment:
I can’t see where this hook is replaced? Perhaps:

```php
$validatorList = $context['Record']->getCMSValidators();
foreach ($validatorList->getValidators() as $validator) {
    $this->invokeWithExtensions('updateFormValidator', $validator, $controller, $name, $context);
}
```

Reply:
Hmmm, I think when I was originally looking at it, I thought it wasn no longer needed (as there is an extension point in the `getCMSValidators()` method), but you're right - this one is also still useful, and I should leave it there. Thanks!